src/lib/services/feature-toggle-service.ts

```diff
@@ -768,18 +768,30 @@ class FeatureToggleService {
         await this.validateName(newFeatureName);
 
         const cToggle =
-            await this.featureStrategiesStore.getFeatureToggleWithEnvs(
+            await this.featureStrategiesStore.getFeatureToggleWithVariantEnvs(
                 featureName,
             );
 
-        const newToggle = { ...cToggle, name: newFeatureName };
+        const newToggle = {
+            ...cToggle,
+            name: newFeatureName,
+            variants: undefined,
```

Comment:
We do always return the variants for legacy purposes here, so we intentionally yeet them so that the creation doesn't create variants we don't want